be2-scala/src/test/scala/ch/epfl/pop/storage/DbActorNewSuite.scala

```diff
@@ -274,4 +274,159 @@ class DbActorNewSuite extends TestKit(ActorSystem("DbActorNewSuiteActorSystem"))
+
+    val dbActor: AskableActorRef = system.actorOf(Props(DbActorNew(mediatorRef, initialStorage)))
+
+    val ask = dbActor ? DbActorNew.Read(channelName1, MESSAGE.message_id)
+    val answer = Await.result(ask, duration)
+
+    answer shouldBe a[DbActorNew.DbActorReadAck]
+
+    val readMessage: Message = answer.asInstanceOf[DbActorNew.DbActorReadAck].message.get
+
+    readMessage should equal(MESSAGE)
+  }
+
+  test("read does not fail for non-existing message (returns None)"){
+    val channelName1: Channel = Channel(CHANNEL_NAME)
+    val initialStorage: InMemoryStorage = InMemoryStorage()
+
+    val dbActor: AskableActorRef = system.actorOf(Props(DbActorNew(mediatorRef, initialStorage)))
+
+    val ask = dbActor ? DbActorNew.Read(channelName1, MESSAGE.message_id)
+    val answer = Await.result(ask, duration)
+
+    answer shouldBe a[DbActorNew.DbActorReadAck]
+
+    answer.asInstanceOf[DbActorNew.DbActorReadAck].message should equal(None)
+  }
+
+  test("readChannelData succeeds for existing ChannelData"){
+    val channelName1: Channel = Channel(CHANNEL_NAME)
+    val initialStorage: InMemoryStorage = InMemoryStorage()
+    val channelData: ChannelData = ChannelData(ObjectType.LAO, Nil)
+    initialStorage.write((CHANNEL_NAME, channelData.toJsonString))
+
+    val dbActor: AskableActorRef = system.actorOf(Props(DbActorNew(mediatorRef, initialStorage)))
+
+    val ask = dbActor ? DbActorNew.ReadChannelData(channelName1)
+    val answer = Await.result(ask, duration)
+
+    answer shouldBe a[DbActorNew.DbActorReadChannelDataAck]
+
+    val readChannelData: ChannelData = answer.asInstanceOf[DbActorNew.DbActorReadChannelDataAck].channelData
+
+    readChannelData should equal(channelData)
+  }
+
+  test("catchup works on a channel with valid ChannelData and messages"){
+    val channelName1: Channel = Channel(CHANNEL_NAME)
+    val initialStorage: InMemoryStorage = InMemoryStorage()
+    val message2 = MESSAGE.copy(message_id = Hash(Base64Data("RmFrZSBtZXNzYWdlX2lkIDopIE5vIGVhc3RlciBlZ2cgcmlnaHQgdGhlcmUhIC0tIE5pY29sYXMgUmF1bGlu")))
+    val listIds: List[Hash] = MESSAGE.message_id :: message2.message_id :: Nil
+    val channelData: ChannelData = ChannelData(ObjectType.LAO, listIds)
+    initialStorage.write(
+      (CHANNEL_NAME, channelData.toJsonString), 
+      (s"$CHANNEL_NAME${Channel.SPECIAL_SEPARATOR}${MESSAGE.message_id}", MESSAGE.toJsonString),
+      (s"$CHANNEL_NAME${Channel.SPECIAL_SEPARATOR}${message2.message_id}", message2.toJsonString),
+    )
+
+    val dbActor: AskableActorRef = system.actorOf(Props(DbActorNew(mediatorRef, initialStorage)))
+
+    val ask = dbActor ? DbActorNew.Catchup(channelName1)
+    val answer = Await.result(ask, duration)
+
+    answer shouldBe a[DbActorNew.DbActorCatchupAck]
+
+    val list: List[Message] = answer.asInstanceOf[DbActorNew.DbActorCatchupAck].messages
+
+    list.size should equal(2)
+    list should contain(MESSAGE)
+    list should contain(message2)
+
```

Comment:
the newlines, in this, the next, and the previous tests do not seem to be consistent :)